examples/dataset_configs.py

```diff
@@ -0,0 +1,137 @@
+from torchvision import transforms
+
+# the various types of dataset found in torchvision
+VISION_DATA_TYPES = {
+    'classification': list(('MNIST FashionMNIST KMNIST EMNIST QMNIST' +
+                            'FakeData LSUM ImageFolder DatasetFolder' +
+                            'ImageNet CIFAR10 CIFAR100 STL10 SVHN' +
+                            'USPS').split(' ')),
+    'captioning': list('CocoCaptions Flickr8k Flickr30k'.split(' ')),
+    #'detection': list('')
+    'segmentation': list('VOCSegmentation SBDataset'.split(' ')),
+    'action_reco': list('UCF101 HMDB51 Kinetics400'.split(' ')),
+}
+
+VISION_COLUMN_NAMES = {
+    'classification': ('image', 'label'),
+    'captioning': ('image', 'caption'),
+    #'detection': None,
+    'segmentation': ('image', 'segmentedImage'),
+    'action_reco': ('video', 'audio', 'label')
+}
+
+
+class DatasetConfig:
+    def __init__(self,
+                 name,
+                 transform=transforms.ToTensor(),
+                 target_transform=None,
+                 train_arg=False,
+                 splits=None,
+                 split_arg_name='split',
+                 downloadable=True,
+                 special_args=None):
+
+        self.name = name
+        self.downloadable = downloadable
+        self.transform = transform
+        self.target_transform = target_transform
+        self.train_arg = train_arg
+        if train_arg:
+            self.splits = ('train', 'test')
+        elif splits is not None:
+            self.splits = splits
+        else:
+            self.splits = None
+        self.split_arg_name = split_arg_name
+        self.special_args = special_args
+
+    def __len__(self):
+        if self.splits is not None:
+            return len(self.splits)
+        else:
+            return 0
+
+    def gen_kwargs(self, root_dir):
+        kwargs = {}
+        kwargs['root'] = root_dir
+        if self.downloadable:
+            kwargs['download'] = True
+        if self.transform:
+            kwargs['transform'] = self.transform
+        if self.target_transform:
+            kwargs['target_transform'] = self.target_transform
+        if self.special_args is not None:
+            kwargs.update(self.special_args)
```

Comment:
This (and some of the code below it) seems unnecessary to me. Since all we are doing is constructing a `kwargs` dict, it shouldn't matter if some are initialized to `None` or other "Falsey" values...

I might be missing some subtletly here, but it seems like the coresponding function which accepts `**kwargs` downstream should be the one which knows how to interpret Falsey values (or just ignore kwargs which it doesn't happen to need). 

Let me know if i'm missing something here though.

Reply:
If the keywords are not accepted by the particular torch dataset's `init()` is raises an error and there is no general configuration for Datasets so we have to make sure we add only the args that the Dataset takes. Thats why I tried to explicitly add keywords to the dict.

Reply:
Ok. I actually wasn't aware of that. 

Just document that behavior and we'll accept it as is :)